src/packages/core/src/charts/parse-signals.ts

```diff
@@ -70,53 +62,37 @@ export default class ParseSignals {
 
   assignSignal() {
     const signals = [];
-    let signal = "";
-
     const datumFormat = this.chartType === 'line' ? 'datum.datum' : 'datum';
 
-    this.tooltipConfig.forEach(field => {
-      if (field.format && field.formatType === 'number' && field.format) {
-        signals.push(`"${field.title}": format(${datumFormat}.${this.stripDatum(field.field)}, "${field.format}")`)
-      } else if (field.format && field.formatType === 'time' && field.format) {
-        signals.push(`"${field.title}": utcFormat(${datumFormat}.${this.stripDatum(field.field)}, "${field.format}")`)
+    this.getTooltipFields().forEach(field => {
+      if (field.format && field.type === 'number') {
+        signals.push(`"${field.property}": format(${datumFormat}.${this.stripDatum(field.column)}, "${field.format}")`);
+      } else if (field.format && field.type === 'date') {
+        signals.push(`"${field.property}": utcFormat(${datumFormat}.${this.stripDatum(field.column)}, "${field.format}")`);
       } else {
-        signals.push(`"${field.title}": ${datumFormat}.${this.stripDatum(field.field)}`)
+        signals.push(`"${field.property}": ${datumFormat}.${this.stripDatum(field.column)}`);
       }
     })
-    
-    signal = `{ ${signals.join()} }`
-    return {
-      signal
-    }
+
+    const signal = `{ ${signals.join()} }`;
+
+    return { signal };
   }
 
   getTooltipFields() {
-    const interactionConfig = this.widgetConfig.hasOwnProperty('interaction_config') ? 
-      this.widgetConfig.interaction_config : 
-      this.schema.hasOwnProperty('interaction_config') ? this.schema.interaction_config : [];
-
-    // We don't have the interaction config object defined
-    if (
-      !interactionConfig ||
-      !interactionConfig.length
-    ) {
-      return [];
+    let interactionConfig = [];
+    if (this.schema.hasOwnProperty('interaction_config')) {
```

Comment:
good fixes, as i said yesterday, i will remove this confusion at some point, so we only have one entry coming in